modAionImpl/src/org/aion/zero/impl/AionHub.java

```diff
@@ -151,72 +168,81 @@ public AionHub() {
-            //noinspection ResultOfMethodCallIgnored
-            rpf.mkdirs();
-            reportsFolder = rpf.getAbsolutePath();
-        }
-
         /*
          * p2p hook up start sync mgr needs to be initialed after loadBlockchain()
          * method
          */
         CfgNetP2p cfgNetP2p = this.cfg.getNet().getP2p();
 
         // there are two p2p implementation , now just point to impl1.
-        this.p2pMgr = new P2pMgr(this.cfg.getNet().getId(), Version.KERNEL_VERSION,
-            this.cfg.getId(), cfgNetP2p.getIp(),
-            cfgNetP2p.getPort(), this.cfg.getNet().getNodes(), cfgNetP2p.getDiscover(),
-            cfgNetP2p.getMaxTempNodes(),
-            cfgNetP2p.getMaxActiveNodes(),
-            cfgNetP2p.getBootlistSyncOnly(), cfgNetP2p.getErrorTolerance());
+        this.p2pMgr =
+                new P2pMgr(
+                        this.cfg.getNet().getId(),
+                        Version.KERNEL_VERSION,
+                        this.cfg.getId(),
+                        cfgNetP2p.getIp(),
+                        cfgNetP2p.getPort(),
+                        this.cfg.getNet().getNodes(),
+                        cfgNetP2p.getDiscover(),
+                        cfgNetP2p.getMaxTempNodes(),
+                        cfgNetP2p.getMaxActiveNodes(),
+                        cfgNetP2p.getBootlistSyncOnly(),
+                        cfgNetP2p.getErrorTolerance());
 
         this.syncMgr = SyncMgr.inst();
-        this.syncMgr.init(this.p2pMgr, this.eventMgr, this.cfg.getSync().getBlocksQueueMax(),
-            this.cfg.getSync().getShowStatus(), this.cfg.getReports().isEnabled(), reportsFolder);
+        this.syncMgr.init(
+                blockchain,
+                p2pMgr,
+                eventMgr,
+                cfg.getSync().getBlocksQueueMax(),
+                cfg.getSync().getShowStatus());
 
         ChainConfiguration chainConfig = new ChainConfiguration();
-        this.propHandler = new BlockPropagationHandler(1024, this.blockchain, this.p2pMgr,
-            chainConfig.createBlockHeaderValidator(), this.cfg.getNet().getP2p().inSyncOnlyMode());
+        this.propHandler =
+                new BlockPropagationHandler(
+                        1024,
+                        this.blockchain,
+                        p2pMgr,
+                        chainConfig.createBlockHeaderValidator(),
+                        cfg.getNet().getP2p().inSyncOnlyMode());
 
         registerCallback();
-        this.p2pMgr.run();
+        p2pMgr.run();
 
         ((AionPendingStateImpl) this.mempool).setP2pMgr(this.p2pMgr);
 
         this.pow = new AionPoW();
         this.pow.init(blockchain, mempool, eventMgr);
+    }
 
-        if (cfg.getReports().isHeapDumpEnabled()) {
-            new Thread(
-                new TaskDumpHeap(this.start, cfg.getReports().getHeapDumpInterval(), reportsFolder),
-                "dump-heap")
-                .start();
-        }
+    static AionHub createForTesting(
```

Comment:
would it be possible to use the @VisibleForTesting annotation for this? i'm not sure whether it can be used on a static method.

Reply:
I think it should work with statics -- it's just an informational annotation